doc/source/_templates/index.html

```diff
@@ -1,153 +1,153 @@
-        aria-controls="v-pills-tuning"
-        aria-selected="false"
-      >
-        Hyperparameter Tuning
-      </a>
-      <a
-        class="nav-link"
-        id="v-pills-serving-tab"
-        data-toggle="pill"
-        role="tab"
-        aria-controls="v-pills-serving"
-        aria-selected="false"
-      >
-        Model Serving
-      </a>
-      <a
-        class="nav-link"
-        id="v-pills-rl-tab"
-        data-toggle="pill"
-        role="tab"
-        aria-controls="v-pills-rl"
-        aria-selected="false"
-      >
-        Reinforcement Learning
-      </a>
-    </div>
-  </div>
-  <div class="col-8">
-    <div class="tab-content" id="v-pills-tabContent">
+  <div class="row">
+    <div class="col-4">
       <div
-        class="tab-pane fade show active"
-        id="v-pills-data"
-        role="tabpanel"
-        aria-labelledby="v-pills-data-tab"
+        class="nav flex-column nav-pills"
+        id="v-pills-tab"
+        aria-orientation="vertical"
       >
-        <pre style="margin: 0">
-          <code class="language-python not-selectable">
+        <a
+          class="nav-link active"
+          id="v-pills-batch-tab"
+          data-toggle="pill"
+          role="tab"
+          aria-controls="v-pills-data"
+          aria-selected="true"
+        >
+          Batch Inference
+        </a>
+        <a
+          class="nav-link"
+          id="v-pills-training-tab"
+          data-toggle="pill"
+          role="tab"
+          aria-controls="v-pills-training"
+          aria-selected="false"
+        >
+          Model Training
+        </a>
+        <a
+          class="nav-link"
+          id="v-pills-tuning-tab"
+          data-toggle="pill"
+          role="tab"
+          aria-controls="v-pills-tuning"
+          aria-selected="false"
+        >
+          Hyperparameter Tuning
+        </a>
+        <a
+          class="nav-link"
+          id="v-pills-serving-tab"
+          data-toggle="pill"
+          role="tab"
+          aria-controls="v-pills-serving"
+          aria-selected="false"
+        >
+          Model Serving
+        </a>
+        <a
+          class="nav-link"
+          id="v-pills-rl-tab"
+          data-toggle="pill"
+          role="tab"
+          aria-controls="v-pills-rl"
+          aria-selected="false"
+        >
+          Reinforcement Learning
+        </a>
+      </div>
+    </div>
+    <div class="col-8">
+      <div class="tab-content" id="v-pills-tabContent">
+        <!-- prettier-ignore -->
+        <div
+          class="tab-pane fade show active"
+          id="v-pills-data"
+          role="tabpanel"
+          aria-labelledby="v-pills-data-tab"
+        >
+        {{ pygments_highlight_python("""
 from typing import Dict
 import numpy as np
 
 import ray
 
 # Step 1: Create a Ray Dataset from in-memory Numpy arrays.
-ds = ray.data.from_numpy(np.asarray(["Complete this", "for me"]))
+ds = ray.data.from_numpy(np.asarray(['Complete this', 'for me']))
```

Comment:
why are we making these changes? seems unrelated to the change required here.

if it has to (or if you want to), could you split them into another PR?

Reply:
These changes are being made automatically because we have a `.pre-commit-config.yml` that runs `prettier`, ensuring that we have consistent code style for the html/css/js that makes up the docs. Previously, we had no code formatting or linting for the docs.

Since we introduced the `pre-commit` config relatively recently, we haven't run it everywhere because we didn't want to touch all files at once. Eventually we will replace `format.sh` with `pre-commit`.

One of the side effects of this is that we are gradually introducing consistent code styling into the docs; any time we touch a file and `prettier` runs, we get changes like this. IMO this approach is less jarring to begin with because pre-commit hooks don't reformat the entire repository, but there's an argument to be made for just making one large commit with the pre-commit hooks run on all files at the same time.

Reply:
could you have a separate PR that:

- Checkin all changes required for passing `pre-commit run -a`
- Add `pre-commit run -a` as part of the premerge lint check"

we do not want PR's to have these unrelated changes that distracts the actual content.

Reply:
I'd really like to have this as a premerge lint check, but we aren't enforcing it repo-wide yet.

So the reason it's this way is that we're in an awkward phase right now where we are poised to move to pre-commit, but still only actively enforcing `format-sh`. And sadly, `format.sh` does nothing for anything in the `docs/` directory :/

I'll update this PR to remove any pre-commit related changes, then disable pre-commit temporarily for this. Whenever ray-ci is ready to make the transition away from `format.sh`, we can run `pre-commit` to get `docs/` formatted.

Reply:
@aslonnie Actually, looking more closely at this particular section: I need these quotes to be `'` and not `"` due to the fact that we are supplying python code to the pygments lexer from an HTML template. We actually need this to be single quoted for this to work - otherwise the templating engine gets confused.

Alternatively, it might work if we use triple `'''` to enclose the python code that gets passed to pygments, allowing us to keep `"` in the code snippet. Would that be preferable?

Reply:
yeah, probably keep it as `"`, I think ray repo's styling uses and enforced `"` for strings everywhere. the code here should be kept consistent.

Reply:
Hmm, well some of the example python code here already uses `'`, so even on `master` we're not consistent with the rest of the python code base. In any case I'll make it all black-compliant.